test/distributed/_tensor/test_dtensor_ops.py

```diff
@@ -141,7 +141,6 @@ def wrapped(fn):
     xfail("diag_embed"),
     xfail("diagflat"),
     xfail("diagonal"),
```

Comment:
Looks like `diagonal_copy` come from `diagonal` so we could probably remove `xfail("diagonal")` as well?
https://github.com/pytorch/pytorch/blob/main/torch/_refs/__init__.py#L6319

Reply:
I wonder why the `xfail` is not giving "Unexpected success" on the unit test though 🤔

Reply:
> I wonder why the `xfail` is not giving "Unexpected success" on the unit test though 🤔

Oh. I think it is because we didn't register op strategy for `aten.diagonal` so it is still failing as expected, but I guess this doesn't need to be included in this PR.

Reply:
ah makes sense!
let me try to add `diagonal` and `diag_embed` in a follow-up